pkg/chrootarchive/archive.go

```diff
@@ -35,13 +48,17 @@ func Untar(tarArchive io.Reader, dest string, options *archive.TarOptions) error
 		}
 	}
 
-	decompressedArchive, err := archive.DecompressStream(tarArchive)
-	if err != nil {
-		return err
+	r := ioutil.NopCloser(tarArchive)
```

Comment:
I agree with @vbatts. If a method is passed an `io.Reader`, it should not be expected to close the reader, since ownership can be assumed is held by the caller. If an `io.ReadCloser` is passed, the caller passes ownership to the callee, which should always invoke close.

Please confirm this convention is followed.